### PR TITLE
Test Query Transaction Reference Link

### PR DESCRIPTION
**Pull request by maektwain**
screen shot 2018-04-08 at 3 56 10 pm

### CODE REVIEW

File: src/components/message.js [hunk 1]
```diff
@@ -20,14 +23,13 @@ function showMessage(type, content, duration) {
...
       msgHandlers[msgKey]();
-
       handleClose();
     }
   };
 
   const msgBody = (
     <span>
-      {content}
+      {content.message} <a href={content.link}>Transaction Reference</a>
```
Comment (cryptomental) [inline src/components/message.js]: Aha! That does the trick.

File: src/components/DeployContract/DeployContractForm.js [hunk 2]
```diff
@@ -43,10 +43,10 @@ class DeployContractForm extends Component {
 
   render() {
     const props = this.computeChildrenProps();
-    const mode = props.mode;    
+    const mode = props.mode;
 
-    return (mode === "guided") ? this.getGuidedDeploymentComponent(props) 
-      : (mode === "quick") ? this.getQuickDeploymentComponent(props) 
+    return (mode === "guided") ? this.getGuidedDeploymentComponent(props)
+      : (mode === "quick") ? this.getQuickDeploymentComponent(props)
```
Comment (cryptomental) [inline src/components/DeployContract/DeployContractForm.js]: I guess all the whitespace was automatically corrected by IDE.

File: src/components/message.js [hunk 3]
```diff
@@ -10,6 +12,7 @@ const iconStyle = {
 const msgHandlers = {};
 
 function showMessage(type, content, duration) {
+  console.log(content);
```
Comment (cryptomental) [inline src/components/message.js]: This can be removed.

File: src/components/message.js (hunk 1 above)
Comment (maektwain) [reply src/components/message.js]: Yeah had to pass objects to get the link working.

File: src/components/TestQuery/TestQueryForm.js [hunk 4]
```diff
@@ -26,16 +26,16 @@ const message = (network, hash) => {
 
   switch(network) {
     case 'rinkeby':
-      message = message + `https://rinkeby.etherscan.io/tx/${hash}`;
+      message = {message: message ,link:`https://rinkeby.etherscan.io/tx/${hash}`};
```
Comment (cryptomental) [inline src/components/TestQuery/TestQueryForm.js]: Minor: inconsistent spacing, '{message: message ,link' should become '{message: message, link'

File: src/components/message.js (hunk 1 above)
Comment (cryptomental) [reply src/components/message.js]: Ok, great, just minor changes and can be merged.

**Comment by pelsasser** [comment]
@cryptomental - thanks for the thorough review!

@maektwain - please make the changes requested by @cryptomental.  Lets change the link text to "View on Etherscan"

once this is all set we will get it merged! @maektwain thanks for all your contributions!

**Comment by eswarasai** [comment]
@maektwain -- Can you commit the suggested changes and also rebase with upstream master to resolve the conflicts? Thanks

File: src/components/TestQuery/TestQueryForm.js (hunk 4 above)
Comment (maektwain) [reply src/components/TestQuery/TestQueryForm.js]: Done

File: src/components/message.js (hunk 3 above)
Comment (maektwain) [reply src/components/message.js]: Done

File: src/components/message.js [hunk 5]
```diff
@@ -1,6 +1,8 @@
 import { Icon, message } from 'antd';
 import React from 'react';
 
+
+
```
[the remark this answers is not in the document]
Comment (maektwain) [reply src/components/message.js]: Done

**Comment by maektwain** [comment]
We can close this one , I have created another PR #93

**Pull request closed without merging**